// locutus/lua/math/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.type = exports.sqrt = exports.sin = exports.min = exports.max = exports.floor = exports.cos = exports.ceil = exports.abs = void 0;
var abs_ts_1 = require("./abs.js");
Object.defineProperty(exports, "abs", { enumerable: true, get: function () { return abs_ts_1.abs; } });
var ceil_ts_1 = require("./ceil.js");
Object.defineProperty(exports, "ceil", { enumerable: true, get: function () { return ceil_ts_1.ceil; } });
var cos_ts_1 = require("./cos.js");
Object.defineProperty(exports, "cos", { enumerable: true, get: function () { return cos_ts_1.cos; } });
var floor_ts_1 = require("./floor.js");
Object.defineProperty(exports, "floor", { enumerable: true, get: function () { return floor_ts_1.floor; } });
var max_ts_1 = require("./max.js");
Object.defineProperty(exports, "max", { enumerable: true, get: function () { return max_ts_1.max; } });
var min_ts_1 = require("./min.js");
Object.defineProperty(exports, "min", { enumerable: true, get: function () { return min_ts_1.min; } });
var sin_ts_1 = require("./sin.js");
Object.defineProperty(exports, "sin", { enumerable: true, get: function () { return sin_ts_1.sin; } });
var sqrt_ts_1 = require("./sqrt.js");
Object.defineProperty(exports, "sqrt", { enumerable: true, get: function () { return sqrt_ts_1.sqrt; } });
var type_ts_1 = require("./type.js");
Object.defineProperty(exports, "type", { enumerable: true, get: function () { return type_ts_1.type; } });
