// locutus/c/ctype/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toupper = exports.tolower = exports.isupper = exports.isspace = exports.islower = exports.isdigit = exports.isalpha = exports.isalnum = void 0;
var isalnum_ts_1 = require("./isalnum.js");
Object.defineProperty(exports, "isalnum", { enumerable: true, get: function () { return isalnum_ts_1.isalnum; } });
var isalpha_ts_1 = require("./isalpha.js");
Object.defineProperty(exports, "isalpha", { enumerable: true, get: function () { return isalpha_ts_1.isalpha; } });
var isdigit_ts_1 = require("./isdigit.js");
Object.defineProperty(exports, "isdigit", { enumerable: true, get: function () { return isdigit_ts_1.isdigit; } });
var islower_ts_1 = require("./islower.js");
Object.defineProperty(exports, "islower", { enumerable: true, get: function () { return islower_ts_1.islower; } });
var isspace_ts_1 = require("./isspace.js");
Object.defineProperty(exports, "isspace", { enumerable: true, get: function () { return isspace_ts_1.isspace; } });
var isupper_ts_1 = require("./isupper.js");
Object.defineProperty(exports, "isupper", { enumerable: true, get: function () { return isupper_ts_1.isupper; } });
var tolower_ts_1 = require("./tolower.js");
Object.defineProperty(exports, "tolower", { enumerable: true, get: function () { return tolower_ts_1.tolower; } });
var toupper_ts_1 = require("./toupper.js");
Object.defineProperty(exports, "toupper", { enumerable: true, get: function () { return toupper_ts_1.toupper; } });
