// locutus/php/filesystem/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.realpath = exports.pathinfo = exports.file_get_contents = exports.file_exists = exports.dirname = exports.basename = void 0;
var basename_ts_1 = require("./basename.js");
Object.defineProperty(exports, "basename", { enumerable: true, get: function () { return basename_ts_1.basename; } });
var dirname_ts_1 = require("./dirname.js");
Object.defineProperty(exports, "dirname", { enumerable: true, get: function () { return dirname_ts_1.dirname; } });
var file_exists_ts_1 = require("./file_exists.js");
Object.defineProperty(exports, "file_exists", { enumerable: true, get: function () { return file_exists_ts_1.file_exists; } });
var file_get_contents_ts_1 = require("./file_get_contents.js");
Object.defineProperty(exports, "file_get_contents", { enumerable: true, get: function () { return file_get_contents_ts_1.file_get_contents; } });
var pathinfo_ts_1 = require("./pathinfo.js");
Object.defineProperty(exports, "pathinfo", { enumerable: true, get: function () { return pathinfo_ts_1.pathinfo; } });
var realpath_ts_1 = require("./realpath.js");
Object.defineProperty(exports, "realpath", { enumerable: true, get: function () { return realpath_ts_1.realpath; } });
