// locutus/python/math/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.trunc = exports.tanh = exports.tan = exports.sqrt = exports.sinh = exports.sin = exports.radians = exports.prod = exports.pow = exports.log2 = exports.log10 = exports.log = exports.lcm = exports.isqrt = exports.isnan = exports.isinf = exports.isfinite = exports.isclose = exports.hypot = exports.gcd = exports.fsum = exports.floor = exports.factorial = exports.fabs = exports.expm1 = exports.exp = exports.dist = exports.degrees = exports.cosh = exports.cos = exports.copysign = exports.comb = exports.ceil = exports.atanh = exports.atan2 = exports.atan = exports.asinh = exports.asin = exports.acosh = exports.acos = void 0;
var acos_ts_1 = require("./acos.js");
Object.defineProperty(exports, "acos", { enumerable: true, get: function () { return acos_ts_1.acos; } });
var acosh_ts_1 = require("./acosh.js");
Object.defineProperty(exports, "acosh", { enumerable: true, get: function () { return acosh_ts_1.acosh; } });
var asin_ts_1 = require("./asin.js");
Object.defineProperty(exports, "asin", { enumerable: true, get: function () { return asin_ts_1.asin; } });
var asinh_ts_1 = require("./asinh.js");
Object.defineProperty(exports, "asinh", { enumerable: true, get: function () { return asinh_ts_1.asinh; } });
var atan_ts_1 = require("./atan.js");
Object.defineProperty(exports, "atan", { enumerable: true, get: function () { return atan_ts_1.atan; } });
var atan2_ts_1 = require("./atan2.js");
Object.defineProperty(exports, "atan2", { enumerable: true, get: function () { return atan2_ts_1.atan2; } });
var atanh_ts_1 = require("./atanh.js");
Object.defineProperty(exports, "atanh", { enumerable: true, get: function () { return atanh_ts_1.atanh; } });
var ceil_ts_1 = require("./ceil.js");
Object.defineProperty(exports, "ceil", { enumerable: true, get: function () { return ceil_ts_1.ceil; } });
var comb_ts_1 = require("./comb.js");
Object.defineProperty(exports, "comb", { enumerable: true, get: function () { return comb_ts_1.comb; } });
var copysign_ts_1 = require("./copysign.js");
Object.defineProperty(exports, "copysign", { enumerable: true, get: function () { return copysign_ts_1.copysign; } });
var cos_ts_1 = require("./cos.js");
Object.defineProperty(exports, "cos", { enumerable: true, get: function () { return cos_ts_1.cos; } });
var cosh_ts_1 = require("./cosh.js");
Object.defineProperty(exports, "cosh", { enumerable: true, get: function () { return cosh_ts_1.cosh; } });
var degrees_ts_1 = require("./degrees.js");
Object.defineProperty(exports, "degrees", { enumerable: true, get: function () { return degrees_ts_1.degrees; } });
var dist_ts_1 = require("./dist.js");
Object.defineProperty(exports, "dist", { enumerable: true, get: function () { return dist_ts_1.dist; } });
var exp_ts_1 = require("./exp.js");
Object.defineProperty(exports, "exp", { enumerable: true, get: function () { return exp_ts_1.exp; } });
var expm1_ts_1 = require("./expm1.js");
Object.defineProperty(exports, "expm1", { enumerable: true, get: function () { return expm1_ts_1.expm1; } });
var fabs_ts_1 = require("./fabs.js");
Object.defineProperty(exports, "fabs", { enumerable: true, get: function () { return fabs_ts_1.fabs; } });
var factorial_ts_1 = require("./factorial.js");
Object.defineProperty(exports, "factorial", { enumerable: true, get: function () { return factorial_ts_1.factorial; } });
var floor_ts_1 = require("./floor.js");
Object.defineProperty(exports, "floor", { enumerable: true, get: function () { return floor_ts_1.floor; } });
var fsum_ts_1 = require("./fsum.js");
Object.defineProperty(exports, "fsum", { enumerable: true, get: function () { return fsum_ts_1.fsum; } });
var gcd_ts_1 = require("./gcd.js");
Object.defineProperty(exports, "gcd", { enumerable: true, get: function () { return gcd_ts_1.gcd; } });
var hypot_ts_1 = require("./hypot.js");
Object.defineProperty(exports, "hypot", { enumerable: true, get: function () { return hypot_ts_1.hypot; } });
var isclose_ts_1 = require("./isclose.js");
Object.defineProperty(exports, "isclose", { enumerable: true, get: function () { return isclose_ts_1.isclose; } });
var isfinite_ts_1 = require("./isfinite.js");
Object.defineProperty(exports, "isfinite", { enumerable: true, get: function () { return isfinite_ts_1.isfinite; } });
var isinf_ts_1 = require("./isinf.js");
Object.defineProperty(exports, "isinf", { enumerable: true, get: function () { return isinf_ts_1.isinf; } });
var isnan_ts_1 = require("./isnan.js");
Object.defineProperty(exports, "isnan", { enumerable: true, get: function () { return isnan_ts_1.isnan; } });
var isqrt_ts_1 = require("./isqrt.js");
Object.defineProperty(exports, "isqrt", { enumerable: true, get: function () { return isqrt_ts_1.isqrt; } });
var lcm_ts_1 = require("./lcm.js");
Object.defineProperty(exports, "lcm", { enumerable: true, get: function () { return lcm_ts_1.lcm; } });
var log_ts_1 = require("./log.js");
Object.defineProperty(exports, "log", { enumerable: true, get: function () { return log_ts_1.log; } });
var log10_ts_1 = require("./log10.js");
Object.defineProperty(exports, "log10", { enumerable: true, get: function () { return log10_ts_1.log10; } });
var log2_ts_1 = require("./log2.js");
Object.defineProperty(exports, "log2", { enumerable: true, get: function () { return log2_ts_1.log2; } });
var pow_ts_1 = require("./pow.js");
Object.defineProperty(exports, "pow", { enumerable: true, get: function () { return pow_ts_1.pow; } });
var prod_ts_1 = require("./prod.js");
Object.defineProperty(exports, "prod", { enumerable: true, get: function () { return prod_ts_1.prod; } });
var radians_ts_1 = require("./radians.js");
Object.defineProperty(exports, "radians", { enumerable: true, get: function () { return radians_ts_1.radians; } });
var sin_ts_1 = require("./sin.js");
Object.defineProperty(exports, "sin", { enumerable: true, get: function () { return sin_ts_1.sin; } });
var sinh_ts_1 = require("./sinh.js");
Object.defineProperty(exports, "sinh", { enumerable: true, get: function () { return sinh_ts_1.sinh; } });
var sqrt_ts_1 = require("./sqrt.js");
Object.defineProperty(exports, "sqrt", { enumerable: true, get: function () { return sqrt_ts_1.sqrt; } });
var tan_ts_1 = require("./tan.js");
Object.defineProperty(exports, "tan", { enumerable: true, get: function () { return tan_ts_1.tan; } });
var tanh_ts_1 = require("./tanh.js");
Object.defineProperty(exports, "tanh", { enumerable: true, get: function () { return tanh_ts_1.tanh; } });
var trunc_ts_1 = require("./trunc.js");
Object.defineProperty(exports, "trunc", { enumerable: true, get: function () { return trunc_ts_1.trunc; } });
