// locutus/ruby/String/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.upcase = exports.tr = exports.strip = exports.start_with = exports.reverse = exports.length = exports.include = exports.gsub = exports.end_with = exports.downcase = exports.delete_suffix = exports.delete_prefix = exports.chop = exports.chomp = exports.capitalize = void 0;
var capitalize_ts_1 = require("./capitalize.js");
Object.defineProperty(exports, "capitalize", { enumerable: true, get: function () { return capitalize_ts_1.capitalize; } });
var chomp_ts_1 = require("./chomp.js");
Object.defineProperty(exports, "chomp", { enumerable: true, get: function () { return chomp_ts_1.chomp; } });
var chop_ts_1 = require("./chop.js");
Object.defineProperty(exports, "chop", { enumerable: true, get: function () { return chop_ts_1.chop; } });
var delete_prefix_ts_1 = require("./delete_prefix.js");
Object.defineProperty(exports, "delete_prefix", { enumerable: true, get: function () { return delete_prefix_ts_1.delete_prefix; } });
var delete_suffix_ts_1 = require("./delete_suffix.js");
Object.defineProperty(exports, "delete_suffix", { enumerable: true, get: function () { return delete_suffix_ts_1.delete_suffix; } });
var downcase_ts_1 = require("./downcase.js");
Object.defineProperty(exports, "downcase", { enumerable: true, get: function () { return downcase_ts_1.downcase; } });
var end_with_ts_1 = require("./end_with.js");
Object.defineProperty(exports, "end_with", { enumerable: true, get: function () { return end_with_ts_1.end_with; } });
var gsub_ts_1 = require("./gsub.js");
Object.defineProperty(exports, "gsub", { enumerable: true, get: function () { return gsub_ts_1.gsub; } });
var include_ts_1 = require("./include.js");
Object.defineProperty(exports, "include", { enumerable: true, get: function () { return include_ts_1.include; } });
var length_ts_1 = require("./length.js");
Object.defineProperty(exports, "length", { enumerable: true, get: function () { return length_ts_1.length; } });
var reverse_ts_1 = require("./reverse.js");
Object.defineProperty(exports, "reverse", { enumerable: true, get: function () { return reverse_ts_1.reverse; } });
var start_with_ts_1 = require("./start_with.js");
Object.defineProperty(exports, "start_with", { enumerable: true, get: function () { return start_with_ts_1.start_with; } });
var strip_ts_1 = require("./strip.js");
Object.defineProperty(exports, "strip", { enumerable: true, get: function () { return strip_ts_1.strip; } });
var tr_ts_1 = require("./tr.js");
Object.defineProperty(exports, "tr", { enumerable: true, get: function () { return tr_ts_1.tr; } });
var upcase_ts_1 = require("./upcase.js");
Object.defineProperty(exports, "upcase", { enumerable: true, get: function () { return upcase_ts_1.upcase; } });
