// locutus/awk/builtin/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toupper = exports.tolower = exports.substr = exports.sqrt = exports.sin = exports.log = exports.length = exports.int = exports.exp = exports.cos = void 0;
var cos_ts_1 = require("./cos.js");
Object.defineProperty(exports, "cos", { enumerable: true, get: function () { return cos_ts_1.cos; } });
var exp_ts_1 = require("./exp.js");
Object.defineProperty(exports, "exp", { enumerable: true, get: function () { return exp_ts_1.exp; } });
var int_ts_1 = require("./int.js");
Object.defineProperty(exports, "int", { enumerable: true, get: function () { return int_ts_1.int; } });
var length_ts_1 = require("./length.js");
Object.defineProperty(exports, "length", { enumerable: true, get: function () { return length_ts_1.length; } });
var log_ts_1 = require("./log.js");
Object.defineProperty(exports, "log", { enumerable: true, get: function () { return log_ts_1.log; } });
var sin_ts_1 = require("./sin.js");
Object.defineProperty(exports, "sin", { enumerable: true, get: function () { return sin_ts_1.sin; } });
var sqrt_ts_1 = require("./sqrt.js");
Object.defineProperty(exports, "sqrt", { enumerable: true, get: function () { return sqrt_ts_1.sqrt; } });
var substr_ts_1 = require("./substr.js");
Object.defineProperty(exports, "substr", { enumerable: true, get: function () { return substr_ts_1.substr; } });
var tolower_ts_1 = require("./tolower.js");
Object.defineProperty(exports, "tolower", { enumerable: true, get: function () { return tolower_ts_1.tolower; } });
var toupper_ts_1 = require("./toupper.js");
Object.defineProperty(exports, "toupper", { enumerable: true, get: function () { return toupper_ts_1.toupper; } });
