// locutus/ruby/Math/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tanh = exports.tan = exports.sqrt = exports.sinh = exports.sin = exports.log2 = exports.log10 = exports.log = exports.exp = exports.cosh = exports.cos = exports.cbrt = exports.atan = exports.asin = exports.acos = void 0;
var acos_ts_1 = require("./acos.js");
Object.defineProperty(exports, "acos", { enumerable: true, get: function () { return acos_ts_1.acos; } });
var asin_ts_1 = require("./asin.js");
Object.defineProperty(exports, "asin", { enumerable: true, get: function () { return asin_ts_1.asin; } });
var atan_ts_1 = require("./atan.js");
Object.defineProperty(exports, "atan", { enumerable: true, get: function () { return atan_ts_1.atan; } });
var cbrt_ts_1 = require("./cbrt.js");
Object.defineProperty(exports, "cbrt", { enumerable: true, get: function () { return cbrt_ts_1.cbrt; } });
var cos_ts_1 = require("./cos.js");
Object.defineProperty(exports, "cos", { enumerable: true, get: function () { return cos_ts_1.cos; } });
var cosh_ts_1 = require("./cosh.js");
Object.defineProperty(exports, "cosh", { enumerable: true, get: function () { return cosh_ts_1.cosh; } });
var exp_ts_1 = require("./exp.js");
Object.defineProperty(exports, "exp", { enumerable: true, get: function () { return exp_ts_1.exp; } });
var log_ts_1 = require("./log.js");
Object.defineProperty(exports, "log", { enumerable: true, get: function () { return log_ts_1.log; } });
var log10_ts_1 = require("./log10.js");
Object.defineProperty(exports, "log10", { enumerable: true, get: function () { return log10_ts_1.log10; } });
var log2_ts_1 = require("./log2.js");
Object.defineProperty(exports, "log2", { enumerable: true, get: function () { return log2_ts_1.log2; } });
var sin_ts_1 = require("./sin.js");
Object.defineProperty(exports, "sin", { enumerable: true, get: function () { return sin_ts_1.sin; } });
var sinh_ts_1 = require("./sinh.js");
Object.defineProperty(exports, "sinh", { enumerable: true, get: function () { return sinh_ts_1.sinh; } });
var sqrt_ts_1 = require("./sqrt.js");
Object.defineProperty(exports, "sqrt", { enumerable: true, get: function () { return sqrt_ts_1.sqrt; } });
var tan_ts_1 = require("./tan.js");
Object.defineProperty(exports, "tan", { enumerable: true, get: function () { return tan_ts_1.tan; } });
var tanh_ts_1 = require("./tanh.js");
Object.defineProperty(exports, "tanh", { enumerable: true, get: function () { return tanh_ts_1.tanh; } });
