// locutus/php/bc/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bcsub = exports.bcscale = exports.bcround = exports.bcmul = exports.bcdiv = exports.bccomp = exports.bcadd = void 0;
var bcadd_ts_1 = require("./bcadd.js");
Object.defineProperty(exports, "bcadd", { enumerable: true, get: function () { return bcadd_ts_1.bcadd; } });
var bccomp_ts_1 = require("./bccomp.js");
Object.defineProperty(exports, "bccomp", { enumerable: true, get: function () { return bccomp_ts_1.bccomp; } });
var bcdiv_ts_1 = require("./bcdiv.js");
Object.defineProperty(exports, "bcdiv", { enumerable: true, get: function () { return bcdiv_ts_1.bcdiv; } });
var bcmul_ts_1 = require("./bcmul.js");
Object.defineProperty(exports, "bcmul", { enumerable: true, get: function () { return bcmul_ts_1.bcmul; } });
var bcround_ts_1 = require("./bcround.js");
Object.defineProperty(exports, "bcround", { enumerable: true, get: function () { return bcround_ts_1.bcround; } });
var bcscale_ts_1 = require("./bcscale.js");
Object.defineProperty(exports, "bcscale", { enumerable: true, get: function () { return bcscale_ts_1.bcscale; } });
var bcsub_ts_1 = require("./bcsub.js");
Object.defineProperty(exports, "bcsub", { enumerable: true, get: function () { return bcsub_ts_1.bcsub; } });
