// locutus/php/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.xml = exports.xdiff = exports.url = exports.strings = exports.pcre = exports.network = exports.misc = exports.math = exports.json = exports.info = exports.i18n = exports.funchand = exports.filesystem = exports.exec = exports.datetime = exports.ctype = exports.bc = exports.array = void 0;
exports.array = __importStar(require("./array/index.js"));
exports.bc = __importStar(require("./bc/index.js"));
exports.ctype = __importStar(require("./ctype/index.js"));
exports.datetime = __importStar(require("./datetime/index.js"));
exports.exec = __importStar(require("./exec/index.js"));
exports.filesystem = __importStar(require("./filesystem/index.js"));
exports.funchand = __importStar(require("./funchand/index.js"));
exports.i18n = __importStar(require("./i18n/index.js"));
exports.info = __importStar(require("./info/index.js"));
exports.json = __importStar(require("./json/index.js"));
exports.math = __importStar(require("./math/index.js"));
exports.misc = __importStar(require("./misc/index.js"));
exports.network = __importStar(require("./network/index.js"));
exports.pcre = __importStar(require("./pcre/index.js"));
exports.strings = __importStar(require("./strings/index.js"));
exports.url = __importStar(require("./url/index.js"));
exports.xdiff = __importStar(require("./xdiff/index.js"));
exports.xml = __importStar(require("./xml/index.js"));
