// locutus/perl/core/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ucfirst = exports.uc = exports.substr = exports.sprintf = exports.split = exports.rindex = exports.reverse = exports.quotemeta = exports.pack = exports.ord = exports.length = exports.lcfirst = exports.lc = exports.join = exports.int = exports.hex = exports.chr = exports.chop = exports.chomp = exports.abs = void 0;
var abs_ts_1 = require("./abs.js");
Object.defineProperty(exports, "abs", { enumerable: true, get: function () { return abs_ts_1.abs; } });
var chomp_ts_1 = require("./chomp.js");
Object.defineProperty(exports, "chomp", { enumerable: true, get: function () { return chomp_ts_1.chomp; } });
var chop_ts_1 = require("./chop.js");
Object.defineProperty(exports, "chop", { enumerable: true, get: function () { return chop_ts_1.chop; } });
var chr_ts_1 = require("./chr.js");
Object.defineProperty(exports, "chr", { enumerable: true, get: function () { return chr_ts_1.chr; } });
var hex_ts_1 = require("./hex.js");
Object.defineProperty(exports, "hex", { enumerable: true, get: function () { return hex_ts_1.hex; } });
var int_ts_1 = require("./int.js");
Object.defineProperty(exports, "int", { enumerable: true, get: function () { return int_ts_1.int; } });
var join_ts_1 = require("./join.js");
Object.defineProperty(exports, "join", { enumerable: true, get: function () { return join_ts_1.join; } });
var lc_ts_1 = require("./lc.js");
Object.defineProperty(exports, "lc", { enumerable: true, get: function () { return lc_ts_1.lc; } });
var lcfirst_ts_1 = require("./lcfirst.js");
Object.defineProperty(exports, "lcfirst", { enumerable: true, get: function () { return lcfirst_ts_1.lcfirst; } });
var length_ts_1 = require("./length.js");
Object.defineProperty(exports, "length", { enumerable: true, get: function () { return length_ts_1.length; } });
var ord_ts_1 = require("./ord.js");
Object.defineProperty(exports, "ord", { enumerable: true, get: function () { return ord_ts_1.ord; } });
var pack_ts_1 = require("./pack.js");
Object.defineProperty(exports, "pack", { enumerable: true, get: function () { return pack_ts_1.pack; } });
var quotemeta_ts_1 = require("./quotemeta.js");
Object.defineProperty(exports, "quotemeta", { enumerable: true, get: function () { return quotemeta_ts_1.quotemeta; } });
var reverse_ts_1 = require("./reverse.js");
Object.defineProperty(exports, "reverse", { enumerable: true, get: function () { return reverse_ts_1.reverse; } });
var rindex_ts_1 = require("./rindex.js");
Object.defineProperty(exports, "rindex", { enumerable: true, get: function () { return rindex_ts_1.rindex; } });
var split_ts_1 = require("./split.js");
Object.defineProperty(exports, "split", { enumerable: true, get: function () { return split_ts_1.split; } });
var sprintf_ts_1 = require("./sprintf.js");
Object.defineProperty(exports, "sprintf", { enumerable: true, get: function () { return sprintf_ts_1.sprintf; } });
var substr_ts_1 = require("./substr.js");
Object.defineProperty(exports, "substr", { enumerable: true, get: function () { return substr_ts_1.substr; } });
var uc_ts_1 = require("./uc.js");
Object.defineProperty(exports, "uc", { enumerable: true, get: function () { return uc_ts_1.uc; } });
var ucfirst_ts_1 = require("./ucfirst.js");
Object.defineProperty(exports, "ucfirst", { enumerable: true, get: function () { return ucfirst_ts_1.ucfirst; } });
