// locutus/tcl/string/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.trimright = exports.trimleft = exports.trim = exports.toupper = exports.tolower = exports.reverse = exports.repeat = exports.regsub = exports.range = exports.match = exports.map = exports.length = exports.last = exports.first = exports.equal = exports.compare = void 0;
var compare_ts_1 = require("./compare.js");
Object.defineProperty(exports, "compare", { enumerable: true, get: function () { return compare_ts_1.compare; } });
var equal_ts_1 = require("./equal.js");
Object.defineProperty(exports, "equal", { enumerable: true, get: function () { return equal_ts_1.equal; } });
var first_ts_1 = require("./first.js");
Object.defineProperty(exports, "first", { enumerable: true, get: function () { return first_ts_1.first; } });
var last_ts_1 = require("./last.js");
Object.defineProperty(exports, "last", { enumerable: true, get: function () { return last_ts_1.last; } });
var length_ts_1 = require("./length.js");
Object.defineProperty(exports, "length", { enumerable: true, get: function () { return length_ts_1.length; } });
var map_ts_1 = require("./map.js");
Object.defineProperty(exports, "map", { enumerable: true, get: function () { return map_ts_1.map; } });
var match_ts_1 = require("./match.js");
Object.defineProperty(exports, "match", { enumerable: true, get: function () { return match_ts_1.match; } });
var range_ts_1 = require("./range.js");
Object.defineProperty(exports, "range", { enumerable: true, get: function () { return range_ts_1.range; } });
var regsub_ts_1 = require("./regsub.js");
Object.defineProperty(exports, "regsub", { enumerable: true, get: function () { return regsub_ts_1.regsub; } });
var repeat_ts_1 = require("./repeat.js");
Object.defineProperty(exports, "repeat", { enumerable: true, get: function () { return repeat_ts_1.repeat; } });
var reverse_ts_1 = require("./reverse.js");
Object.defineProperty(exports, "reverse", { enumerable: true, get: function () { return reverse_ts_1.reverse; } });
var tolower_ts_1 = require("./tolower.js");
Object.defineProperty(exports, "tolower", { enumerable: true, get: function () { return tolower_ts_1.tolower; } });
var toupper_ts_1 = require("./toupper.js");
Object.defineProperty(exports, "toupper", { enumerable: true, get: function () { return toupper_ts_1.toupper; } });
var trim_ts_1 = require("./trim.js");
Object.defineProperty(exports, "trim", { enumerable: true, get: function () { return trim_ts_1.trim; } });
var trimleft_ts_1 = require("./trimleft.js");
Object.defineProperty(exports, "trimleft", { enumerable: true, get: function () { return trimleft_ts_1.trimleft; } });
var trimright_ts_1 = require("./trimright.js");
Object.defineProperty(exports, "trimright", { enumerable: true, get: function () { return trimright_ts_1.trimright; } });
