// locutus/php/var/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.var_export = exports.var_dump = exports.unserialize = exports.strval = exports.serialize = exports.print_r = exports.isset = exports.is_unicode = exports.is_string = exports.is_scalar = exports.is_real = exports.is_object = exports.is_numeric = exports.is_null = exports.is_long = exports.is_integer = exports.is_int = exports.is_float = exports.is_double = exports.is_callable = exports.is_buffer = exports.is_bool = exports.is_binary = exports.is_array = exports.intval = exports.gettype = exports.floatval = exports.empty = exports.doubleval = exports.boolval = void 0;
var boolval_ts_1 = require("./boolval.js");
Object.defineProperty(exports, "boolval", { enumerable: true, get: function () { return boolval_ts_1.boolval; } });
var doubleval_ts_1 = require("./doubleval.js");
Object.defineProperty(exports, "doubleval", { enumerable: true, get: function () { return doubleval_ts_1.doubleval; } });
var empty_ts_1 = require("./empty.js");
Object.defineProperty(exports, "empty", { enumerable: true, get: function () { return empty_ts_1.empty; } });
var floatval_ts_1 = require("./floatval.js");
Object.defineProperty(exports, "floatval", { enumerable: true, get: function () { return floatval_ts_1.floatval; } });
var gettype_ts_1 = require("./gettype.js");
Object.defineProperty(exports, "gettype", { enumerable: true, get: function () { return gettype_ts_1.gettype; } });
var intval_ts_1 = require("./intval.js");
Object.defineProperty(exports, "intval", { enumerable: true, get: function () { return intval_ts_1.intval; } });
var is_array_ts_1 = require("./is_array.js");
Object.defineProperty(exports, "is_array", { enumerable: true, get: function () { return is_array_ts_1.is_array; } });
var is_binary_ts_1 = require("./is_binary.js");
Object.defineProperty(exports, "is_binary", { enumerable: true, get: function () { return is_binary_ts_1.is_binary; } });
var is_bool_ts_1 = require("./is_bool.js");
Object.defineProperty(exports, "is_bool", { enumerable: true, get: function () { return is_bool_ts_1.is_bool; } });
var is_buffer_ts_1 = require("./is_buffer.js");
Object.defineProperty(exports, "is_buffer", { enumerable: true, get: function () { return is_buffer_ts_1.is_buffer; } });
var is_callable_ts_1 = require("./is_callable.js");
Object.defineProperty(exports, "is_callable", { enumerable: true, get: function () { return is_callable_ts_1.is_callable; } });
var is_double_ts_1 = require("./is_double.js");
Object.defineProperty(exports, "is_double", { enumerable: true, get: function () { return is_double_ts_1.is_double; } });
var is_float_ts_1 = require("./is_float.js");
Object.defineProperty(exports, "is_float", { enumerable: true, get: function () { return is_float_ts_1.is_float; } });
var is_int_ts_1 = require("./is_int.js");
Object.defineProperty(exports, "is_int", { enumerable: true, get: function () { return is_int_ts_1.is_int; } });
var is_integer_ts_1 = require("./is_integer.js");
Object.defineProperty(exports, "is_integer", { enumerable: true, get: function () { return is_integer_ts_1.is_integer; } });
var is_long_ts_1 = require("./is_long.js");
Object.defineProperty(exports, "is_long", { enumerable: true, get: function () { return is_long_ts_1.is_long; } });
var is_null_ts_1 = require("./is_null.js");
Object.defineProperty(exports, "is_null", { enumerable: true, get: function () { return is_null_ts_1.is_null; } });
var is_numeric_ts_1 = require("./is_numeric.js");
Object.defineProperty(exports, "is_numeric", { enumerable: true, get: function () { return is_numeric_ts_1.is_numeric; } });
var is_object_ts_1 = require("./is_object.js");
Object.defineProperty(exports, "is_object", { enumerable: true, get: function () { return is_object_ts_1.is_object; } });
var is_real_ts_1 = require("./is_real.js");
Object.defineProperty(exports, "is_real", { enumerable: true, get: function () { return is_real_ts_1.is_real; } });
var is_scalar_ts_1 = require("./is_scalar.js");
Object.defineProperty(exports, "is_scalar", { enumerable: true, get: function () { return is_scalar_ts_1.is_scalar; } });
var is_string_ts_1 = require("./is_string.js");
Object.defineProperty(exports, "is_string", { enumerable: true, get: function () { return is_string_ts_1.is_string; } });
var is_unicode_ts_1 = require("./is_unicode.js");
Object.defineProperty(exports, "is_unicode", { enumerable: true, get: function () { return is_unicode_ts_1.is_unicode; } });
var isset_ts_1 = require("./isset.js");
Object.defineProperty(exports, "isset", { enumerable: true, get: function () { return isset_ts_1.isset; } });
var print_r_ts_1 = require("./print_r.js");
Object.defineProperty(exports, "print_r", { enumerable: true, get: function () { return print_r_ts_1.print_r; } });
var serialize_ts_1 = require("./serialize.js");
Object.defineProperty(exports, "serialize", { enumerable: true, get: function () { return serialize_ts_1.serialize; } });
var strval_ts_1 = require("./strval.js");
Object.defineProperty(exports, "strval", { enumerable: true, get: function () { return strval_ts_1.strval; } });
var unserialize_ts_1 = require("./unserialize.js");
Object.defineProperty(exports, "unserialize", { enumerable: true, get: function () { return unserialize_ts_1.unserialize; } });
var var_dump_ts_1 = require("./var_dump.js");
Object.defineProperty(exports, "var_dump", { enumerable: true, get: function () { return var_dump_ts_1.var_dump; } });
var var_export_ts_1 = require("./var_export.js");
Object.defineProperty(exports, "var_export", { enumerable: true, get: function () { return var_export_ts_1.var_export; } });
