// locutus/lua/string/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.upper = exports.sub = exports.reverse = exports.rep = exports.lower = exports.len = exports.gsub = void 0;
var gsub_ts_1 = require("./gsub.js");
Object.defineProperty(exports, "gsub", { enumerable: true, get: function () { return gsub_ts_1.gsub; } });
var len_ts_1 = require("./len.js");
Object.defineProperty(exports, "len", { enumerable: true, get: function () { return len_ts_1.len; } });
var lower_ts_1 = require("./lower.js");
Object.defineProperty(exports, "lower", { enumerable: true, get: function () { return lower_ts_1.lower; } });
var rep_ts_1 = require("./rep.js");
Object.defineProperty(exports, "rep", { enumerable: true, get: function () { return rep_ts_1.rep; } });
var reverse_ts_1 = require("./reverse.js");
Object.defineProperty(exports, "reverse", { enumerable: true, get: function () { return reverse_ts_1.reverse; } });
var sub_ts_1 = require("./sub.js");
Object.defineProperty(exports, "sub", { enumerable: true, get: function () { return sub_ts_1.sub; } });
var upper_ts_1 = require("./upper.js");
Object.defineProperty(exports, "upper", { enumerable: true, get: function () { return upper_ts_1.upper; } });
