// locutus/php/strings/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.str_repeat = exports.str_pad = exports.str_ireplace = exports.str_getcsv = exports.sscanf = exports.sprintf = exports.split = exports.soundex = exports.similar_text = exports.sha1_file = exports.sha1 = exports.setlocale = exports.rtrim = exports.quotemeta = exports.quoted_printable_encode = exports.quoted_printable_decode = exports.printf = exports.parse_str = exports.ord = exports.number_format = exports.nl_langinfo = exports.nl2br = exports.money_format = exports.metaphone = exports.md5_file = exports.md5 = exports.ltrim = exports.localeconv = exports.levenshtein = exports.lcfirst = exports.join = exports.implode = exports.htmlspecialchars_decode = exports.htmlspecialchars = exports.htmlentities = exports.html_entity_decode = exports.hex2bin = exports.get_html_translation_table = exports.explode = exports.echo = exports.crc32 = exports.count_chars = exports.convert_uuencode = exports.convert_cyr_string = exports.chunk_split = exports.chr = exports.chop = exports.bin2hex = exports.addslashes = exports.addcslashes = void 0;
exports.wordwrap = exports.vsprintf = exports.vprintf = exports.ucwords = exports.ucfirst = exports.trim = exports.substr_replace = exports.substr_count = exports.substr_compare = exports.substr = exports.strtr = exports.strtoupper = exports.strtolower = exports.strtok = exports.strstr = exports.strspn = exports.strrpos = exports.strripos = exports.strrev = exports.strrchr = exports.strpos = exports.strpbrk = exports.strncmp = exports.strncasecmp = exports.strnatcmp = exports.strnatcasecmp = exports.strlen = exports.stristr = exports.stripslashes = exports.stripos = exports.strip_tags = exports.strcspn = exports.strcoll = exports.strcmp = exports.strchr = exports.strcasecmp = exports.str_word_count = exports.str_split = exports.str_shuffle = exports.str_rot13 = exports.str_replace = void 0;
var addcslashes_ts_1 = require("./addcslashes.js");
Object.defineProperty(exports, "addcslashes", { enumerable: true, get: function () { return addcslashes_ts_1.addcslashes; } });
var addslashes_ts_1 = require("./addslashes.js");
Object.defineProperty(exports, "addslashes", { enumerable: true, get: function () { return addslashes_ts_1.addslashes; } });
var bin2hex_ts_1 = require("./bin2hex.js");
Object.defineProperty(exports, "bin2hex", { enumerable: true, get: function () { return bin2hex_ts_1.bin2hex; } });
var chop_ts_1 = require("./chop.js");
Object.defineProperty(exports, "chop", { enumerable: true, get: function () { return chop_ts_1.chop; } });
var chr_ts_1 = require("./chr.js");
Object.defineProperty(exports, "chr", { enumerable: true, get: function () { return chr_ts_1.chr; } });
var chunk_split_ts_1 = require("./chunk_split.js");
Object.defineProperty(exports, "chunk_split", { enumerable: true, get: function () { return chunk_split_ts_1.chunk_split; } });
var convert_cyr_string_ts_1 = require("./convert_cyr_string.js");
Object.defineProperty(exports, "convert_cyr_string", { enumerable: true, get: function () { return convert_cyr_string_ts_1.convert_cyr_string; } });
var convert_uuencode_ts_1 = require("./convert_uuencode.js");
Object.defineProperty(exports, "convert_uuencode", { enumerable: true, get: function () { return convert_uuencode_ts_1.convert_uuencode; } });
var count_chars_ts_1 = require("./count_chars.js");
Object.defineProperty(exports, "count_chars", { enumerable: true, get: function () { return count_chars_ts_1.count_chars; } });
var crc32_ts_1 = require("./crc32.js");
Object.defineProperty(exports, "crc32", { enumerable: true, get: function () { return crc32_ts_1.crc32; } });
var echo_ts_1 = require("./echo.js");
Object.defineProperty(exports, "echo", { enumerable: true, get: function () { return echo_ts_1.echo; } });
var explode_ts_1 = require("./explode.js");
Object.defineProperty(exports, "explode", { enumerable: true, get: function () { return explode_ts_1.explode; } });
var get_html_translation_table_ts_1 = require("./get_html_translation_table.js");
Object.defineProperty(exports, "get_html_translation_table", { enumerable: true, get: function () { return get_html_translation_table_ts_1.get_html_translation_table; } });
var hex2bin_ts_1 = require("./hex2bin.js");
Object.defineProperty(exports, "hex2bin", { enumerable: true, get: function () { return hex2bin_ts_1.hex2bin; } });
var html_entity_decode_ts_1 = require("./html_entity_decode.js");
Object.defineProperty(exports, "html_entity_decode", { enumerable: true, get: function () { return html_entity_decode_ts_1.html_entity_decode; } });
var htmlentities_ts_1 = require("./htmlentities.js");
Object.defineProperty(exports, "htmlentities", { enumerable: true, get: function () { return htmlentities_ts_1.htmlentities; } });
var htmlspecialchars_ts_1 = require("./htmlspecialchars.js");
Object.defineProperty(exports, "htmlspecialchars", { enumerable: true, get: function () { return htmlspecialchars_ts_1.htmlspecialchars; } });
var htmlspecialchars_decode_ts_1 = require("./htmlspecialchars_decode.js");
Object.defineProperty(exports, "htmlspecialchars_decode", { enumerable: true, get: function () { return htmlspecialchars_decode_ts_1.htmlspecialchars_decode; } });
var implode_ts_1 = require("./implode.js");
Object.defineProperty(exports, "implode", { enumerable: true, get: function () { return implode_ts_1.implode; } });
var join_ts_1 = require("./join.js");
Object.defineProperty(exports, "join", { enumerable: true, get: function () { return join_ts_1.join; } });
var lcfirst_ts_1 = require("./lcfirst.js");
Object.defineProperty(exports, "lcfirst", { enumerable: true, get: function () { return lcfirst_ts_1.lcfirst; } });
var levenshtein_ts_1 = require("./levenshtein.js");
Object.defineProperty(exports, "levenshtein", { enumerable: true, get: function () { return levenshtein_ts_1.levenshtein; } });
var localeconv_ts_1 = require("./localeconv.js");
Object.defineProperty(exports, "localeconv", { enumerable: true, get: function () { return localeconv_ts_1.localeconv; } });
var ltrim_ts_1 = require("./ltrim.js");
Object.defineProperty(exports, "ltrim", { enumerable: true, get: function () { return ltrim_ts_1.ltrim; } });
var md5_ts_1 = require("./md5.js");
Object.defineProperty(exports, "md5", { enumerable: true, get: function () { return md5_ts_1.md5; } });
var md5_file_ts_1 = require("./md5_file.js");
Object.defineProperty(exports, "md5_file", { enumerable: true, get: function () { return md5_file_ts_1.md5_file; } });
var metaphone_ts_1 = require("./metaphone.js");
Object.defineProperty(exports, "metaphone", { enumerable: true, get: function () { return metaphone_ts_1.metaphone; } });
var money_format_ts_1 = require("./money_format.js");
Object.defineProperty(exports, "money_format", { enumerable: true, get: function () { return money_format_ts_1.money_format; } });
var nl2br_ts_1 = require("./nl2br.js");
Object.defineProperty(exports, "nl2br", { enumerable: true, get: function () { return nl2br_ts_1.nl2br; } });
var nl_langinfo_ts_1 = require("./nl_langinfo.js");
Object.defineProperty(exports, "nl_langinfo", { enumerable: true, get: function () { return nl_langinfo_ts_1.nl_langinfo; } });
var number_format_ts_1 = require("./number_format.js");
Object.defineProperty(exports, "number_format", { enumerable: true, get: function () { return number_format_ts_1.number_format; } });
var ord_ts_1 = require("./ord.js");
Object.defineProperty(exports, "ord", { enumerable: true, get: function () { return ord_ts_1.ord; } });
var parse_str_ts_1 = require("./parse_str.js");
Object.defineProperty(exports, "parse_str", { enumerable: true, get: function () { return parse_str_ts_1.parse_str; } });
var printf_ts_1 = require("./printf.js");
Object.defineProperty(exports, "printf", { enumerable: true, get: function () { return printf_ts_1.printf; } });
var quoted_printable_decode_ts_1 = require("./quoted_printable_decode.js");
Object.defineProperty(exports, "quoted_printable_decode", { enumerable: true, get: function () { return quoted_printable_decode_ts_1.quoted_printable_decode; } });
var quoted_printable_encode_ts_1 = require("./quoted_printable_encode.js");
Object.defineProperty(exports, "quoted_printable_encode", { enumerable: true, get: function () { return quoted_printable_encode_ts_1.quoted_printable_encode; } });
var quotemeta_ts_1 = require("./quotemeta.js");
Object.defineProperty(exports, "quotemeta", { enumerable: true, get: function () { return quotemeta_ts_1.quotemeta; } });
var rtrim_ts_1 = require("./rtrim.js");
Object.defineProperty(exports, "rtrim", { enumerable: true, get: function () { return rtrim_ts_1.rtrim; } });
var setlocale_ts_1 = require("./setlocale.js");
Object.defineProperty(exports, "setlocale", { enumerable: true, get: function () { return setlocale_ts_1.setlocale; } });
var sha1_ts_1 = require("./sha1.js");
Object.defineProperty(exports, "sha1", { enumerable: true, get: function () { return sha1_ts_1.sha1; } });
var sha1_file_ts_1 = require("./sha1_file.js");
Object.defineProperty(exports, "sha1_file", { enumerable: true, get: function () { return sha1_file_ts_1.sha1_file; } });
var similar_text_ts_1 = require("./similar_text.js");
Object.defineProperty(exports, "similar_text", { enumerable: true, get: function () { return similar_text_ts_1.similar_text; } });
var soundex_ts_1 = require("./soundex.js");
Object.defineProperty(exports, "soundex", { enumerable: true, get: function () { return soundex_ts_1.soundex; } });
var split_ts_1 = require("./split.js");
Object.defineProperty(exports, "split", { enumerable: true, get: function () { return split_ts_1.split; } });
var sprintf_ts_1 = require("./sprintf.js");
Object.defineProperty(exports, "sprintf", { enumerable: true, get: function () { return sprintf_ts_1.sprintf; } });
var sscanf_ts_1 = require("./sscanf.js");
Object.defineProperty(exports, "sscanf", { enumerable: true, get: function () { return sscanf_ts_1.sscanf; } });
var str_getcsv_ts_1 = require("./str_getcsv.js");
Object.defineProperty(exports, "str_getcsv", { enumerable: true, get: function () { return str_getcsv_ts_1.str_getcsv; } });
var str_ireplace_ts_1 = require("./str_ireplace.js");
Object.defineProperty(exports, "str_ireplace", { enumerable: true, get: function () { return str_ireplace_ts_1.str_ireplace; } });
var str_pad_ts_1 = require("./str_pad.js");
Object.defineProperty(exports, "str_pad", { enumerable: true, get: function () { return str_pad_ts_1.str_pad; } });
var str_repeat_ts_1 = require("./str_repeat.js");
Object.defineProperty(exports, "str_repeat", { enumerable: true, get: function () { return str_repeat_ts_1.str_repeat; } });
var str_replace_ts_1 = require("./str_replace.js");
Object.defineProperty(exports, "str_replace", { enumerable: true, get: function () { return str_replace_ts_1.str_replace; } });
var str_rot13_ts_1 = require("./str_rot13.js");
Object.defineProperty(exports, "str_rot13", { enumerable: true, get: function () { return str_rot13_ts_1.str_rot13; } });
var str_shuffle_ts_1 = require("./str_shuffle.js");
Object.defineProperty(exports, "str_shuffle", { enumerable: true, get: function () { return str_shuffle_ts_1.str_shuffle; } });
var str_split_ts_1 = require("./str_split.js");
Object.defineProperty(exports, "str_split", { enumerable: true, get: function () { return str_split_ts_1.str_split; } });
var str_word_count_ts_1 = require("./str_word_count.js");
Object.defineProperty(exports, "str_word_count", { enumerable: true, get: function () { return str_word_count_ts_1.str_word_count; } });
var strcasecmp_ts_1 = require("./strcasecmp.js");
Object.defineProperty(exports, "strcasecmp", { enumerable: true, get: function () { return strcasecmp_ts_1.strcasecmp; } });
var strchr_ts_1 = require("./strchr.js");
Object.defineProperty(exports, "strchr", { enumerable: true, get: function () { return strchr_ts_1.strchr; } });
var strcmp_ts_1 = require("./strcmp.js");
Object.defineProperty(exports, "strcmp", { enumerable: true, get: function () { return strcmp_ts_1.strcmp; } });
var strcoll_ts_1 = require("./strcoll.js");
Object.defineProperty(exports, "strcoll", { enumerable: true, get: function () { return strcoll_ts_1.strcoll; } });
var strcspn_ts_1 = require("./strcspn.js");
Object.defineProperty(exports, "strcspn", { enumerable: true, get: function () { return strcspn_ts_1.strcspn; } });
var strip_tags_ts_1 = require("./strip_tags.js");
Object.defineProperty(exports, "strip_tags", { enumerable: true, get: function () { return strip_tags_ts_1.strip_tags; } });
var stripos_ts_1 = require("./stripos.js");
Object.defineProperty(exports, "stripos", { enumerable: true, get: function () { return stripos_ts_1.stripos; } });
var stripslashes_ts_1 = require("./stripslashes.js");
Object.defineProperty(exports, "stripslashes", { enumerable: true, get: function () { return stripslashes_ts_1.stripslashes; } });
var stristr_ts_1 = require("./stristr.js");
Object.defineProperty(exports, "stristr", { enumerable: true, get: function () { return stristr_ts_1.stristr; } });
var strlen_ts_1 = require("./strlen.js");
Object.defineProperty(exports, "strlen", { enumerable: true, get: function () { return strlen_ts_1.strlen; } });
var strnatcasecmp_ts_1 = require("./strnatcasecmp.js");
Object.defineProperty(exports, "strnatcasecmp", { enumerable: true, get: function () { return strnatcasecmp_ts_1.strnatcasecmp; } });
var strnatcmp_ts_1 = require("./strnatcmp.js");
Object.defineProperty(exports, "strnatcmp", { enumerable: true, get: function () { return strnatcmp_ts_1.strnatcmp; } });
var strncasecmp_ts_1 = require("./strncasecmp.js");
Object.defineProperty(exports, "strncasecmp", { enumerable: true, get: function () { return strncasecmp_ts_1.strncasecmp; } });
var strncmp_ts_1 = require("./strncmp.js");
Object.defineProperty(exports, "strncmp", { enumerable: true, get: function () { return strncmp_ts_1.strncmp; } });
var strpbrk_ts_1 = require("./strpbrk.js");
Object.defineProperty(exports, "strpbrk", { enumerable: true, get: function () { return strpbrk_ts_1.strpbrk; } });
var strpos_ts_1 = require("./strpos.js");
Object.defineProperty(exports, "strpos", { enumerable: true, get: function () { return strpos_ts_1.strpos; } });
var strrchr_ts_1 = require("./strrchr.js");
Object.defineProperty(exports, "strrchr", { enumerable: true, get: function () { return strrchr_ts_1.strrchr; } });
var strrev_ts_1 = require("./strrev.js");
Object.defineProperty(exports, "strrev", { enumerable: true, get: function () { return strrev_ts_1.strrev; } });
var strripos_ts_1 = require("./strripos.js");
Object.defineProperty(exports, "strripos", { enumerable: true, get: function () { return strripos_ts_1.strripos; } });
var strrpos_ts_1 = require("./strrpos.js");
Object.defineProperty(exports, "strrpos", { enumerable: true, get: function () { return strrpos_ts_1.strrpos; } });
var strspn_ts_1 = require("./strspn.js");
Object.defineProperty(exports, "strspn", { enumerable: true, get: function () { return strspn_ts_1.strspn; } });
var strstr_ts_1 = require("./strstr.js");
Object.defineProperty(exports, "strstr", { enumerable: true, get: function () { return strstr_ts_1.strstr; } });
var strtok_ts_1 = require("./strtok.js");
Object.defineProperty(exports, "strtok", { enumerable: true, get: function () { return strtok_ts_1.strtok; } });
var strtolower_ts_1 = require("./strtolower.js");
Object.defineProperty(exports, "strtolower", { enumerable: true, get: function () { return strtolower_ts_1.strtolower; } });
var strtoupper_ts_1 = require("./strtoupper.js");
Object.defineProperty(exports, "strtoupper", { enumerable: true, get: function () { return strtoupper_ts_1.strtoupper; } });
var strtr_ts_1 = require("./strtr.js");
Object.defineProperty(exports, "strtr", { enumerable: true, get: function () { return strtr_ts_1.strtr; } });
var substr_ts_1 = require("./substr.js");
Object.defineProperty(exports, "substr", { enumerable: true, get: function () { return substr_ts_1.substr; } });
var substr_compare_ts_1 = require("./substr_compare.js");
Object.defineProperty(exports, "substr_compare", { enumerable: true, get: function () { return substr_compare_ts_1.substr_compare; } });
var substr_count_ts_1 = require("./substr_count.js");
Object.defineProperty(exports, "substr_count", { enumerable: true, get: function () { return substr_count_ts_1.substr_count; } });
var substr_replace_ts_1 = require("./substr_replace.js");
Object.defineProperty(exports, "substr_replace", { enumerable: true, get: function () { return substr_replace_ts_1.substr_replace; } });
var trim_ts_1 = require("./trim.js");
Object.defineProperty(exports, "trim", { enumerable: true, get: function () { return trim_ts_1.trim; } });
var ucfirst_ts_1 = require("./ucfirst.js");
Object.defineProperty(exports, "ucfirst", { enumerable: true, get: function () { return ucfirst_ts_1.ucfirst; } });
var ucwords_ts_1 = require("./ucwords.js");
Object.defineProperty(exports, "ucwords", { enumerable: true, get: function () { return ucwords_ts_1.ucwords; } });
var vprintf_ts_1 = require("./vprintf.js");
Object.defineProperty(exports, "vprintf", { enumerable: true, get: function () { return vprintf_ts_1.vprintf; } });
var vsprintf_ts_1 = require("./vsprintf.js");
Object.defineProperty(exports, "vsprintf", { enumerable: true, get: function () { return vsprintf_ts_1.vsprintf; } });
var wordwrap_ts_1 = require("./wordwrap.js");
Object.defineProperty(exports, "wordwrap", { enumerable: true, get: function () { return wordwrap_ts_1.wordwrap; } });
