// locutus/php/info/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.version_compare = exports.set_time_limit = exports.ini_set = exports.ini_get = exports.getenv = exports.assert_options = void 0;
var assert_options_ts_1 = require("./assert_options.js");
Object.defineProperty(exports, "assert_options", { enumerable: true, get: function () { return assert_options_ts_1.assert_options; } });
var getenv_ts_1 = require("./getenv.js");
Object.defineProperty(exports, "getenv", { enumerable: true, get: function () { return getenv_ts_1.getenv; } });
var ini_get_ts_1 = require("./ini_get.js");
Object.defineProperty(exports, "ini_get", { enumerable: true, get: function () { return ini_get_ts_1.ini_get; } });
var ini_set_ts_1 = require("./ini_set.js");
Object.defineProperty(exports, "ini_set", { enumerable: true, get: function () { return ini_set_ts_1.ini_set; } });
var set_time_limit_ts_1 = require("./set_time_limit.js");
Object.defineProperty(exports, "set_time_limit", { enumerable: true, get: function () { return set_time_limit_ts_1.set_time_limit; } });
var version_compare_ts_1 = require("./version_compare.js");
Object.defineProperty(exports, "version_compare", { enumerable: true, get: function () { return version_compare_ts_1.version_compare; } });
