// locutus/powershell/string/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.trimstart = exports.trimend = exports.trim = exports.toupper = exports.tolower = exports.substring = exports.startswith = exports.split = exports.replace = exports.remove = exports.padright = exports.padleft = exports.length = exports.lastindexof = exports.insert = exports.indexof = exports.endswith = exports.contains = void 0;
var contains_ts_1 = require("./contains.js");
Object.defineProperty(exports, "contains", { enumerable: true, get: function () { return contains_ts_1.contains; } });
var endswith_ts_1 = require("./endswith.js");
Object.defineProperty(exports, "endswith", { enumerable: true, get: function () { return endswith_ts_1.endswith; } });
var indexof_ts_1 = require("./indexof.js");
Object.defineProperty(exports, "indexof", { enumerable: true, get: function () { return indexof_ts_1.indexof; } });
var insert_ts_1 = require("./insert.js");
Object.defineProperty(exports, "insert", { enumerable: true, get: function () { return insert_ts_1.insert; } });
var lastindexof_ts_1 = require("./lastindexof.js");
Object.defineProperty(exports, "lastindexof", { enumerable: true, get: function () { return lastindexof_ts_1.lastindexof; } });
var length_ts_1 = require("./length.js");
Object.defineProperty(exports, "length", { enumerable: true, get: function () { return length_ts_1.length; } });
var padleft_ts_1 = require("./padleft.js");
Object.defineProperty(exports, "padleft", { enumerable: true, get: function () { return padleft_ts_1.padleft; } });
var padright_ts_1 = require("./padright.js");
Object.defineProperty(exports, "padright", { enumerable: true, get: function () { return padright_ts_1.padright; } });
var remove_ts_1 = require("./remove.js");
Object.defineProperty(exports, "remove", { enumerable: true, get: function () { return remove_ts_1.remove; } });
var replace_ts_1 = require("./replace.js");
Object.defineProperty(exports, "replace", { enumerable: true, get: function () { return replace_ts_1.replace; } });
var split_ts_1 = require("./split.js");
Object.defineProperty(exports, "split", { enumerable: true, get: function () { return split_ts_1.split; } });
var startswith_ts_1 = require("./startswith.js");
Object.defineProperty(exports, "startswith", { enumerable: true, get: function () { return startswith_ts_1.startswith; } });
var substring_ts_1 = require("./substring.js");
Object.defineProperty(exports, "substring", { enumerable: true, get: function () { return substring_ts_1.substring; } });
var tolower_ts_1 = require("./tolower.js");
Object.defineProperty(exports, "tolower", { enumerable: true, get: function () { return tolower_ts_1.tolower; } });
var toupper_ts_1 = require("./toupper.js");
Object.defineProperty(exports, "toupper", { enumerable: true, get: function () { return toupper_ts_1.toupper; } });
var trim_ts_1 = require("./trim.js");
Object.defineProperty(exports, "trim", { enumerable: true, get: function () { return trim_ts_1.trim; } });
var trimend_ts_1 = require("./trimend.js");
Object.defineProperty(exports, "trimend", { enumerable: true, get: function () { return trimend_ts_1.trimend; } });
var trimstart_ts_1 = require("./trimstart.js");
Object.defineProperty(exports, "trimstart", { enumerable: true, get: function () { return trimstart_ts_1.trimstart; } });
