// locutus/golang/strconv/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Unquote = exports.Quote = exports.ParseInt = exports.ParseFloat = exports.ParseBool = exports.Itoa = exports.FormatInt = exports.FormatFloat = exports.FormatBool = exports.Atoi = void 0;
var Atoi_ts_1 = require("./Atoi.js");
Object.defineProperty(exports, "Atoi", { enumerable: true, get: function () { return Atoi_ts_1.Atoi; } });
var FormatBool_ts_1 = require("./FormatBool.js");
Object.defineProperty(exports, "FormatBool", { enumerable: true, get: function () { return FormatBool_ts_1.FormatBool; } });
var FormatFloat_ts_1 = require("./FormatFloat.js");
Object.defineProperty(exports, "FormatFloat", { enumerable: true, get: function () { return FormatFloat_ts_1.FormatFloat; } });
var FormatInt_ts_1 = require("./FormatInt.js");
Object.defineProperty(exports, "FormatInt", { enumerable: true, get: function () { return FormatInt_ts_1.FormatInt; } });
var Itoa_ts_1 = require("./Itoa.js");
Object.defineProperty(exports, "Itoa", { enumerable: true, get: function () { return Itoa_ts_1.Itoa; } });
var ParseBool_ts_1 = require("./ParseBool.js");
Object.defineProperty(exports, "ParseBool", { enumerable: true, get: function () { return ParseBool_ts_1.ParseBool; } });
var ParseFloat_ts_1 = require("./ParseFloat.js");
Object.defineProperty(exports, "ParseFloat", { enumerable: true, get: function () { return ParseFloat_ts_1.ParseFloat; } });
var ParseInt_ts_1 = require("./ParseInt.js");
Object.defineProperty(exports, "ParseInt", { enumerable: true, get: function () { return ParseInt_ts_1.ParseInt; } });
var Quote_ts_1 = require("./Quote.js");
Object.defineProperty(exports, "Quote", { enumerable: true, get: function () { return Quote_ts_1.Quote; } });
var Unquote_ts_1 = require("./Unquote.js");
Object.defineProperty(exports, "Unquote", { enumerable: true, get: function () { return Unquote_ts_1.Unquote; } });
