// locutus/python/string/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.whitespace = exports.punctuation = exports.printable = exports.octdigits = exports.hexdigits = exports.digits = exports.capwords = exports.ascii_uppercase = exports.ascii_lowercase = exports.ascii_letters = void 0;
var ascii_letters_ts_1 = require("./ascii_letters.js");
Object.defineProperty(exports, "ascii_letters", { enumerable: true, get: function () { return ascii_letters_ts_1.ascii_letters; } });
var ascii_lowercase_ts_1 = require("./ascii_lowercase.js");
Object.defineProperty(exports, "ascii_lowercase", { enumerable: true, get: function () { return ascii_lowercase_ts_1.ascii_lowercase; } });
var ascii_uppercase_ts_1 = require("./ascii_uppercase.js");
Object.defineProperty(exports, "ascii_uppercase", { enumerable: true, get: function () { return ascii_uppercase_ts_1.ascii_uppercase; } });
var capwords_ts_1 = require("./capwords.js");
Object.defineProperty(exports, "capwords", { enumerable: true, get: function () { return capwords_ts_1.capwords; } });
var digits_ts_1 = require("./digits.js");
Object.defineProperty(exports, "digits", { enumerable: true, get: function () { return digits_ts_1.digits; } });
var hexdigits_ts_1 = require("./hexdigits.js");
Object.defineProperty(exports, "hexdigits", { enumerable: true, get: function () { return hexdigits_ts_1.hexdigits; } });
var octdigits_ts_1 = require("./octdigits.js");
Object.defineProperty(exports, "octdigits", { enumerable: true, get: function () { return octdigits_ts_1.octdigits; } });
var printable_ts_1 = require("./printable.js");
Object.defineProperty(exports, "printable", { enumerable: true, get: function () { return printable_ts_1.printable; } });
var punctuation_ts_1 = require("./punctuation.js");
Object.defineProperty(exports, "punctuation", { enumerable: true, get: function () { return punctuation_ts_1.punctuation; } });
var whitespace_ts_1 = require("./whitespace.js");
Object.defineProperty(exports, "whitespace", { enumerable: true, get: function () { return whitespace_ts_1.whitespace; } });
