// locutus/php/math/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tanh = exports.tan = exports.sqrt = exports.sinh = exports.sin = exports.round = exports.rand = exports.rad2deg = exports.pow = exports.pi = exports.octdec = exports.mt_rand = exports.mt_getrandmax = exports.min = exports.max = exports.log1p = exports.log10 = exports.log = exports.lcg_value = exports.is_nan = exports.is_infinite = exports.is_finite = exports.hypot = exports.hexdec = exports.getrandmax = exports.fmod = exports.floor = exports.expm1 = exports.exp = exports.deg2rad = exports.decoct = exports.dechex = exports.decbin = exports.cosh = exports.cos = exports.ceil = exports.bindec = exports.base_convert = exports.atanh = exports.atan2 = exports.atan = exports.asinh = exports.asin = exports.acosh = exports.acos = exports.abs = void 0;
var abs_ts_1 = require("./abs.js");
Object.defineProperty(exports, "abs", { enumerable: true, get: function () { return abs_ts_1.abs; } });
var acos_ts_1 = require("./acos.js");
Object.defineProperty(exports, "acos", { enumerable: true, get: function () { return acos_ts_1.acos; } });
var acosh_ts_1 = require("./acosh.js");
Object.defineProperty(exports, "acosh", { enumerable: true, get: function () { return acosh_ts_1.acosh; } });
var asin_ts_1 = require("./asin.js");
Object.defineProperty(exports, "asin", { enumerable: true, get: function () { return asin_ts_1.asin; } });
var asinh_ts_1 = require("./asinh.js");
Object.defineProperty(exports, "asinh", { enumerable: true, get: function () { return asinh_ts_1.asinh; } });
var atan_ts_1 = require("./atan.js");
Object.defineProperty(exports, "atan", { enumerable: true, get: function () { return atan_ts_1.atan; } });
var atan2_ts_1 = require("./atan2.js");
Object.defineProperty(exports, "atan2", { enumerable: true, get: function () { return atan2_ts_1.atan2; } });
var atanh_ts_1 = require("./atanh.js");
Object.defineProperty(exports, "atanh", { enumerable: true, get: function () { return atanh_ts_1.atanh; } });
var base_convert_ts_1 = require("./base_convert.js");
Object.defineProperty(exports, "base_convert", { enumerable: true, get: function () { return base_convert_ts_1.base_convert; } });
var bindec_ts_1 = require("./bindec.js");
Object.defineProperty(exports, "bindec", { enumerable: true, get: function () { return bindec_ts_1.bindec; } });
var ceil_ts_1 = require("./ceil.js");
Object.defineProperty(exports, "ceil", { enumerable: true, get: function () { return ceil_ts_1.ceil; } });
var cos_ts_1 = require("./cos.js");
Object.defineProperty(exports, "cos", { enumerable: true, get: function () { return cos_ts_1.cos; } });
var cosh_ts_1 = require("./cosh.js");
Object.defineProperty(exports, "cosh", { enumerable: true, get: function () { return cosh_ts_1.cosh; } });
var decbin_ts_1 = require("./decbin.js");
Object.defineProperty(exports, "decbin", { enumerable: true, get: function () { return decbin_ts_1.decbin; } });
var dechex_ts_1 = require("./dechex.js");
Object.defineProperty(exports, "dechex", { enumerable: true, get: function () { return dechex_ts_1.dechex; } });
var decoct_ts_1 = require("./decoct.js");
Object.defineProperty(exports, "decoct", { enumerable: true, get: function () { return decoct_ts_1.decoct; } });
var deg2rad_ts_1 = require("./deg2rad.js");
Object.defineProperty(exports, "deg2rad", { enumerable: true, get: function () { return deg2rad_ts_1.deg2rad; } });
var exp_ts_1 = require("./exp.js");
Object.defineProperty(exports, "exp", { enumerable: true, get: function () { return exp_ts_1.exp; } });
var expm1_ts_1 = require("./expm1.js");
Object.defineProperty(exports, "expm1", { enumerable: true, get: function () { return expm1_ts_1.expm1; } });
var floor_ts_1 = require("./floor.js");
Object.defineProperty(exports, "floor", { enumerable: true, get: function () { return floor_ts_1.floor; } });
var fmod_ts_1 = require("./fmod.js");
Object.defineProperty(exports, "fmod", { enumerable: true, get: function () { return fmod_ts_1.fmod; } });
var getrandmax_ts_1 = require("./getrandmax.js");
Object.defineProperty(exports, "getrandmax", { enumerable: true, get: function () { return getrandmax_ts_1.getrandmax; } });
var hexdec_ts_1 = require("./hexdec.js");
Object.defineProperty(exports, "hexdec", { enumerable: true, get: function () { return hexdec_ts_1.hexdec; } });
var hypot_ts_1 = require("./hypot.js");
Object.defineProperty(exports, "hypot", { enumerable: true, get: function () { return hypot_ts_1.hypot; } });
var is_finite_ts_1 = require("./is_finite.js");
Object.defineProperty(exports, "is_finite", { enumerable: true, get: function () { return is_finite_ts_1.is_finite; } });
var is_infinite_ts_1 = require("./is_infinite.js");
Object.defineProperty(exports, "is_infinite", { enumerable: true, get: function () { return is_infinite_ts_1.is_infinite; } });
var is_nan_ts_1 = require("./is_nan.js");
Object.defineProperty(exports, "is_nan", { enumerable: true, get: function () { return is_nan_ts_1.is_nan; } });
var lcg_value_ts_1 = require("./lcg_value.js");
Object.defineProperty(exports, "lcg_value", { enumerable: true, get: function () { return lcg_value_ts_1.lcg_value; } });
var log_ts_1 = require("./log.js");
Object.defineProperty(exports, "log", { enumerable: true, get: function () { return log_ts_1.log; } });
var log10_ts_1 = require("./log10.js");
Object.defineProperty(exports, "log10", { enumerable: true, get: function () { return log10_ts_1.log10; } });
var log1p_ts_1 = require("./log1p.js");
Object.defineProperty(exports, "log1p", { enumerable: true, get: function () { return log1p_ts_1.log1p; } });
var max_ts_1 = require("./max.js");
Object.defineProperty(exports, "max", { enumerable: true, get: function () { return max_ts_1.max; } });
var min_ts_1 = require("./min.js");
Object.defineProperty(exports, "min", { enumerable: true, get: function () { return min_ts_1.min; } });
var mt_getrandmax_ts_1 = require("./mt_getrandmax.js");
Object.defineProperty(exports, "mt_getrandmax", { enumerable: true, get: function () { return mt_getrandmax_ts_1.mt_getrandmax; } });
var mt_rand_ts_1 = require("./mt_rand.js");
Object.defineProperty(exports, "mt_rand", { enumerable: true, get: function () { return mt_rand_ts_1.mt_rand; } });
var octdec_ts_1 = require("./octdec.js");
Object.defineProperty(exports, "octdec", { enumerable: true, get: function () { return octdec_ts_1.octdec; } });
var pi_ts_1 = require("./pi.js");
Object.defineProperty(exports, "pi", { enumerable: true, get: function () { return pi_ts_1.pi; } });
var pow_ts_1 = require("./pow.js");
Object.defineProperty(exports, "pow", { enumerable: true, get: function () { return pow_ts_1.pow; } });
var rad2deg_ts_1 = require("./rad2deg.js");
Object.defineProperty(exports, "rad2deg", { enumerable: true, get: function () { return rad2deg_ts_1.rad2deg; } });
var rand_ts_1 = require("./rand.js");
Object.defineProperty(exports, "rand", { enumerable: true, get: function () { return rand_ts_1.rand; } });
var round_ts_1 = require("./round.js");
Object.defineProperty(exports, "round", { enumerable: true, get: function () { return round_ts_1.round; } });
var sin_ts_1 = require("./sin.js");
Object.defineProperty(exports, "sin", { enumerable: true, get: function () { return sin_ts_1.sin; } });
var sinh_ts_1 = require("./sinh.js");
Object.defineProperty(exports, "sinh", { enumerable: true, get: function () { return sinh_ts_1.sinh; } });
var sqrt_ts_1 = require("./sqrt.js");
Object.defineProperty(exports, "sqrt", { enumerable: true, get: function () { return sqrt_ts_1.sqrt; } });
var tan_ts_1 = require("./tan.js");
Object.defineProperty(exports, "tan", { enumerable: true, get: function () { return tan_ts_1.tan; } });
var tanh_ts_1 = require("./tanh.js");
Object.defineProperty(exports, "tanh", { enumerable: true, get: function () { return tanh_ts_1.tanh; } });
